packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -175,19 +183,23 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
             return;
         }
         try {
-            let preferences;
-            if (model.valid) {
-                const content = model.getText();
-                preferences = this.getParsedContent(content);
-            } else {
-                preferences = {};
-            }
-            this.handlePreferenceChanges(preferences);
+            const content = model.valid ? model.getText() : '';
+            this.readPreferencesFromContent(content);
         } catch (e) {
             console.error(`Failed to load preferences from '${this.getUri()}'.`, e);
         }
     }
 
+    protected readPreferencesFromContent(content: string): void {
+        let preferences;
+        try {
+            preferences = this.getParsedContent(content);
+        } catch {
+            preferences = {};
+        }
+        this.handlePreferenceChanges(preferences);
```

Comment:
`handlePreferencesChanged` is a misnomer: I think `updatePreferencesTo` would be clearer.

Reply:
Agreed, but I'm not sure it's worth a breaking change for anyone who extends this? I'd be curious to know how many people extend core / core-like services, because in general I'm pretty open to breaking changes if they're any kind of improvement. But we're pretty aggressive about our uplifts, and we keep pretty well on top of breaking changes. It sounds like Arm, for example, is a few versions behind and might be less happy to find a lot of (potentially) breaking changes after a multi-version uplift.